osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -123,44 +161,61 @@ private double computeAimValue()
             // It is important to also consider accuracy difficulty when doing that
             aimValue *= 0.98 + Math.Pow(Attributes.OverallDifficulty, 2) / 2500;
 
+            // Scale by % FC'd to encourage FC.
+            aimValue *= 0.8 + 0.2 * (Score.MaxCombo / (double)Attributes.MaxCombo);
+
+            if (categoryRatings != null)
+            {
+                categoryRatings.Add("Aim", aimValue);
+                categoryRatings.Add("Aim Combo Stars", aimComboStarRating);
+                categoryRatings.Add("Aim Miss Count Stars", aimMissCountStarRating);
+            }
+
             return aimValue;
         }
 
-        private double computeSpeedValue()
+        private double computeSpeedValue(Dictionary<string, double> categoryRatings = null)
         {
-            double speedValue = Math.Pow(5.0 * Math.Max(1.0, Attributes.SpeedStrain / 0.0675) - 4.0, 3.0) / 100000.0;
+            double speedComboStarRating = comboStarRating(Attributes.SpeedComboStarRatings);
+            double speedMissCountStarRating = missCountStarRating(Attributes.SpeedMissCounts, Attributes.SpeedComboStarRatings.Last());
 
-            // Longer maps are worth more
-            double lengthBonus = 0.95 + 0.4 * Math.Min(1.0, totalHits / 2000.0) +
-                                 (totalHits > 2000 ? Math.Log10(totalHits / 2000.0) * 0.5 : 0.0);
-            speedValue *= lengthBonus;
+            double rawSpeed = Math.Pow(speedComboStarRating, combo_weight) * Math.Pow(speedMissCountStarRating, 1 - combo_weight);
 
-            // Penalize misses by assessing # of misses relative to the total # of objects. Default a 3% reduction for any # of misses.
-            if (countMiss > 0)
-                speedValue *= 0.97 * Math.Pow(1 - Math.Pow((double)countMiss / totalHits, 0.775), Math.Pow(countMiss, .875));
+            double speedValue = Math.Pow(5.0f * Math.Max(1.0f, rawSpeed / 0.0675f) - 4.0f, 3.0f) / 100000.0f;
 
-            // Combo scaling
-            if (Attributes.MaxCombo > 0)
-                speedValue *= Math.Min(Math.Pow(scoreMaxCombo, 0.8) / Math.Pow(Attributes.MaxCombo, 0.8), 1.0);
+            // Penalize misses by assessing # of misses relative to the total # of objects.
+            if (effectiveMissCount > 0)
+                speedValue *= Math.Pow(1 - Math.Pow(effectiveMissCount / totalHits, 0.775), effectiveMissCount);
 
             double approachRateFactor = 0.0;
             if (Attributes.ApproachRate > 10.33)
                 approachRateFactor += 0.4 * (Attributes.ApproachRate - 10.33);
 
-            speedValue *= 1.0 + Math.Min(approachRateFactor, approachRateFactor * (totalHits / 1000.0));
+            // We don't scale buff by length for speed since speed isnt sensitive to any memorization techniques.
+            speedValue *= 1.0 + Math.Min(approachRateFactor, approachRateFactor);
```

Comment:
This change is not only nonsensical (does `Math.Min(x, x)`), but also reverts a significant part of https://github.com/ppy/osu/pull/11107. Please do not sneak such changes in larger diffs, it looks very bad/disingenuous. If you want to revert that adjustment, please PR separately.